src/System.Net.HttpListener/src/System/Net/WebSockets/HttpWebSocket.cs

```diff
@@ -81,6 +82,96 @@ internal static string GetSecWebSocketAcceptString(string secWebSocketKey)
                     clientSecWebSocketProtocol,
                     subProtocol));
         }
+
+        internal static void ValidateOptions(string subProtocol, int receiveBufferSize, int sendBufferSize, TimeSpan keepAliveInterval)
+        {
+            if (subProtocol != null)
+            {
+                WebSocketValidate.ValidateSubprotocol(subProtocol);
+            }
+            
+            if (receiveBufferSize < MinReceiveBufferSize)
+            {
+                throw new ArgumentOutOfRangeException(nameof(receiveBufferSize), receiveBufferSize,
+                    SR.Format(SR.net_WebSockets_ArgumentOutOfRange_TooSmall, MinReceiveBufferSize));
+            }
+
+            if (sendBufferSize < MinSendBufferSize)
+            {
+                throw new ArgumentOutOfRangeException(nameof(sendBufferSize), sendBufferSize,
+                    SR.Format(SR.net_WebSockets_ArgumentOutOfRange_TooSmall, MinSendBufferSize));
+            }
+
+            if (receiveBufferSize > MaxBufferSize)
+            {
+                throw new ArgumentOutOfRangeException(nameof(receiveBufferSize), receiveBufferSize,
+                    SR.Format(SR.net_WebSockets_ArgumentOutOfRange_TooBig,
+                        nameof(receiveBufferSize),
+                        receiveBufferSize,
+                        MaxBufferSize));
+            }
+
+            if (sendBufferSize > MaxBufferSize)
+            {
+                throw new ArgumentOutOfRangeException(nameof(sendBufferSize), sendBufferSize,
+                    SR.Format(SR.net_WebSockets_ArgumentOutOfRange_TooBig,
+                        nameof(sendBufferSize),
+                        sendBufferSize,
+                        MaxBufferSize));
+            }
+
+            if (keepAliveInterval < Timeout.InfiniteTimeSpan) // -1 millisecond
+            {
+                throw new ArgumentOutOfRangeException(nameof(keepAliveInterval), keepAliveInterval,
+                    SR.Format(SR.net_WebSockets_ArgumentOutOfRange_TooSmall, Timeout.InfiniteTimeSpan.ToString()));
+            }
+        }
+
+        internal const int MinSendBufferSize = 16;
+        internal const int MinReceiveBufferSize = 256;
+        internal const int MaxBufferSize = 64 * 1024;
+
+        private static void ValidateWebSocketHeaders(HttpListenerContext context)
+        {
+            ValidateWebSocketHeadersCore(context);
+
```

Comment:
This empty line is fine.  There are empty lines between each piece of the validation.